### PR TITLE
Enable CI builds in travis and appveyor.

### PR DESCRIPTION
**Pull request by Connorhd**
- Add Travis CI config for Linux and OS X
- Add AppVeyor config for Windows
- Update tests to pass on all 3 OSs

See https://github.com/ariya/phantomjs/issues/13828 (lots of details in that issue already).

One major caveat, the AppVeyor build will only run on a paid (Pro FOSS) level of account ($29.95 a month) the free tier does not have enough processing power to build in the timeline. The Travis CI builds run fine for free though.

See build outputs here: https://travis-ci.org/Connorhd/phantomjs/builds/127159050 https://ci.appveyor.com/project/Connorhd/phantomjs

### CODE REVIEW

**Comment by ariya** [comment]
Thanks for the excellent work @Connorhd!

We use Travis CI in the past. It worked for a while, but at some point it ceased to work again. I didn't investigate in details but IIRC it was caused by an imposed limitation on build runs, probably as the VM was being adjusted to account for more Travis CI usage. Hence, I'm a bit hesitant to rely on it too much, as it may break again in the future.

As for AppVeyor,  since we don't have any fund for the Pro level, I don't think we should pursue that right now. In fact, perhaps this is an opportunity to lobby AppVeyor if they are willing to sponsor the CI build of PhantomJS.

With that in mind, would you be willing to separate the Travis CI enablement into its own pull request? After that, I'd be happy to merge that one first.

**Comment by Connorhd** [comment]
Sure thing I will split out the changes.

As for the travisci capacity I think the big problem was that `multiprocessing.cpu_count()` returns 32 (when in reality only 2 cores are available). Running with that much parallelism actually caused memory usage to peak over the available limit. It seems fairly stable with `-j 2` so I think its worth trying.

**Comment by ariya** [comment]
I did some tests and the OS X build tends to fail from time to time:
```
The job exceeded the maxmimum time limit for jobs, and has been terminated.
```

**Comment by XhmikosR** [comment]
@ariya: why don't you just use the free AppVeyor plan? I'm having issue #14095 for all my Windows projects and I'd like to get this merged so that we can fix it.

File: appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+version: 1.0.{build}
+image: Visual Studio 2013
```
Comment (XhmikosR) [inline appveyor.yml]: Is there any reason for VS2013 and not the default image, VS2015?

**Comment by Vitallium** [comment]
@XhmikosR we can't use AppVeyour because doesn't meet our requirements. From their documentation - https://www.appveyor.com/docs/build-configuration#virtual-machine-configurations

We can't build PhantomJS with the free plan specification. It will fail.

File: appveyor.yml (hunk 1 above)
Comment (Vitallium) [inline appveyor.yml]: We have 3rd party libraries compiled with MSVC 20123. I recently updated all libraries to VS2015. I think we are safe to use VS2015.

**Comment by XhmikosR** [comment]
@Vitallium: memory-wise or core-wise?

/CC @FeodorFitsner

**Comment by XhmikosR** [comment]
@Vitallium: I see the builds pass on your test account, don't they?

**Comment by Vitallium** [comment]
@XhmikosR memory. We need at least 4 GB to successfully build PhantomJS. But using only 1 core will significantly decrease the build time. WebKit is a huge project. Building it on a 1-core machine is really slow.

**Comment by Vitallium** [comment]
@XhmikosR 
> I see the builds pass on your test account, don't they?
sorry? I didn't use AppVeyor for PhantomJS. Could you please post the link?

**Comment by XhmikosR** [comment]
https://ci.appveyor.com/project/Connorhd/phantomjs/history

**Comment by Vitallium** [comment]
@XhmikosR Thanks, but that's not me :) Anyway, looks like it compiles successfully. I think I need to test it.
But the build time (46 min 59 sec) is really huge.

**Comment by XhmikosR** [comment]
Sorry, wrong CC :P But yeah, the builds seem to pass with the changes in this branch.

The build time is huge, I feel you. It's something I've been bugging @FeodorFitsner. Recently he told me they are going to fix update all VMs to use the same resources, even free plans.

**Comment by Vitallium** [comment]
But we will switch to a prebuilt WebKit soon. So we will use any kind of CI for PhantomJS. 

See this PR: #14386

**Comment by ariya** [comment]
> https://ci.appveyor.com/project/Connorhd/phantomjs/history

Read the discussion carefully, that's not using the free plan, but rather a trial of a pro plan ($30/mo).

**Comment by XhmikosR** [comment]
Just wait a few days and both plans will use the same environment.

**Comment by XhmikosR** [comment]
Can someone try again? AppVeyor should be like 10x faster now.

**Comment by ariya** [comment]
This needs to be recreated now that our focus is on 2.5 (see #14458).

**Pull request closed without merging**